element_miniscope/miniscope.py

```diff
@@ -0,0 +1,849 @@
+        """
+
+    @property
+    def key_source(self):
+        caiman_key_source = (Fluorescence * ActivityExtractionMethod
+                             * ProcessingParamSet.proj('processing_method')
+                             & 'processing_method = "caiman"'
+                             & 'extraction_method LIKE "caiman%"')
+
+        return caiman_key_source.proj()
+
+    def make(self, key):
+        method, loaded_result = get_loader_result(key, Curation)
+
+        if method == 'caiman':
+            loaded_caiman = loaded_result
+
+            if key['extraction_method'] in ('caiman_deconvolution', 'caiman_dff'):
+                attr_mapper = {'caiman_deconvolution': 'spikes', 'caiman_dff': 'dff'}
+
+                # infer `segmentation_channel` from `params` if available, 
+                # else from caiman loader
+                params = (ProcessingParamSet * ProcessingTask & key).fetch1('params')
+                segmentation_channel = params.get('segmentation_channel',
+                                                  loaded_caiman.segmentation_channel)
+
+                self.insert1(key)
+                self.Trace.insert([{**key,
+                                    'mask_id': mask['mask_id'],
+                                    'fluorescence_channel': segmentation_channel,
+                                    'activity_trace': mask[attr_mapper[key['extraction_method']]]}
+                                    for mask in loaded_caiman.masks])
+
+
+        else:
+            raise NotImplementedError('Unknown/unimplemented method: {}'.format(method))
+
+
+# Helper Functions ---------------------------------------------------------------------
+
+_table_attribute_mapper = {'ProcessingTask': 'processing_output_dir',
+                           'Curation': 'curation_output_dir'}
+
+
+def get_loader_result(key, table):
+    """
+    Retrieve the loaded processed imaging results from the loader (e.g. caiman, etc.)
+        :param key: the `key` to one entry of ProcessingTask or Curation
+        :param table: the class defining the table to retrieve
+         the loaded results from (e.g. ProcessingTask, Curation)
+        :return: a loader object of the loaded results
+         (e.g. caiman.CaImAn, etc.)
+    """
+    method, output_dir = (ProcessingParamSet * table & key).fetch1(
+        'processing_method', _table_attribute_mapper[table.__name__])
+
+    output_dir = find_full_path(get_miniscope_root_data_dir(), output_dir)
+
+    if method == 'caiman':
+        from element_interface import caiman_loader
+        loaded_output = caiman_loader.CaImAn(output_dir)
+    else:
+        raise NotImplementedError('Unknown/unimplemented method: {}'.format(method))
+
+    return method, loaded_output
+
+
+def populate_all(display_progress=True):
+
+    populate_settings = {'display_progress': display_progress, 
+                         'reserve_jobs': False, 
+                         'suppress_errors': False}
```

Comment:
To put these within the schema is a departure from convention elsewhere. Do we want this in the schema or in a workflow `process.py` script?

If in the workflow, I would also allow move the `reserve_jobs` and `suppress_errors` to the arguments with the same defaults.
```suggestion
def populate_all(display_progress=True, reserve_jobs=False, suppress_errors=False):

    populate_settings = {'display_progress': display_progress, 
                         'reserve_jobs': reserve_jobs, 
                         'suppress_errors': suppress_errors}
```

Reply:
Thanks @CBroz1.  I would suggest moving this function to the `element` so that it is not duplicated across each users' workflows.  And if you agree, we can then make this update across the other `element` and `workflow` repositories.

Good point regarding the arguments.  I have committed your suggestion.

Reply:
Noted in [datajoint-elements issue 98](https://github.com/datajoint/datajoint-elements/issues/98)